pkg/ingester/ingester_v2.go

```diff
@@ -547,12 +577,24 @@ func (i *Ingester) v2Query(ctx context.Context, req *client.QueryRequest) (*clie
 	}
 	defer q.Close()
 
+	backfillSSs, err := db.backfillSelect(ctx, int64(from), int64(through), matchers)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the sorted series only if we have to merge it with backfill TSDB results.
 	// It's not required to return sorted series because series are sorted by the Cortex querier.
-	ss := q.Select(false, nil, matchers...)
+	ss := q.Select(len(backfillSSs) > 0, nil, matchers...)
 	if ss.Err() != nil {
 		return nil, ss.Err()
 	}
 
+	if len(backfillSSs) > 0 {
+		// TODO(codesome): If any TSDB in backfill buckets were overlapping
+		// with main TSDB, then use tsdb.NewMergedVerticalSeriesSet
```

Comment:
What's the impact of never using `NewMergedVerticalSeriesSet`?

Reply:
There could be overlapping chunks between the main TSDB and backfill TSDBs, and not vertically merging them might lead to wrong results. I think with recent refactoring in Prometheus, both the logic were merged into one. I will check that and replace stuff accordingly.

Reply:
> I think with recent refactoring in Prometheus, both the logic were merged into one.

That was only for chunks, specific to compaction. So we need to manually decide here whether to use vertical merging or not.

Reply:
If not using the vertical compaction could lead to incorrect results, then sure we should use it. I would always use it and we should be fine.

Reply:
Right, in case of backfill TSDBs it sounds sane to use vertical always.